include/swift/AST/DiagnosticsParse.def

```diff
@@ -762,6 +762,13 @@ ERROR(extra_rbracket,PointsToFirstBadToken,
       "unexpected ']' in type; did you mean to write an array type?", ())
 ERROR(extra_colon,PointsToFirstBadToken,
       "unexpected ':' in type; did you mean to write a dictionary type?", ())
+WARNING(subscript_array_element, none,
+        "unexpected subscript in array literal; did you mean to write two "
```

Comment:
Sorry, I do have some belated comments on the wording of the diagnostic:

Given that it's valid Swift which generates a warning, it's not "unexpected" like a parse error is--and, specifically, the use of a "subscript in array literal" is not at all unexpected, particularly because it's one of the fix-it suggestions!

I'd suggest maybe "expression is a subscript of preceding element; did you mean to write two separate elements?" ("Instead" is unnecessary because essentially every "did you mean" implies it.)